common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/FinalizeShuffleMerge.java

```diff
@@ -68,7 +68,7 @@ public boolean equals(Object other) {
     if (other != null && other instanceof FinalizeShuffleMerge) {
       FinalizeShuffleMerge o = (FinalizeShuffleMerge) other;
       return Objects.equal(appId, o.appId)
-        && appAttemptId == appAttemptId
+        && appAttemptId == o.appAttemptId
```

Comment:
This is introduced via https://github.com/apache/spark/pull/33078/files#diff-a4d872ac22f3cbdfb587125ee9dc4c70fe9efca58e3de5b31dad227f689046f9R71 which landed at master/3.2 4 days ago.

So, we don't need this at branch-3.1.